Source/PartnerCenter.CustomerPortal/BusinessLogic/Commerce/PaymentGateways/PayUGateway.cs

```diff
@@ -0,0 +1,437 @@
+        public PayUGateway(ApplicationDomain applicationDomain, string description) : base(applicationDomain)
+        {
+            description.AssertNotEmpty(nameof(description));
+            this.paymentDescription = description;
+
+            this.payerId = string.Empty;
+            this.paymentId = string.Empty;
+        }
+
+        /// <summary>
+        /// Validates payment configuration. 
+        /// </summary>
+        /// <param name="paymentConfig">The Payment configuration.</param>
+        public void ValidateConfiguration(PaymentConfiguration paymentConfig)
+        {
+            ////Payu does not provide payment profile validation api.
+        }
+
+        /// <summary>
+        /// Creates Web Experience profile using portal branding and payment configuration. 
+        /// </summary>
+        /// <param name="paymentConfig">The Payment configuration.</param>
+        /// <param name="brandConfig">The branding configuration.</param>
+        /// <param name="countryIso2Code">The locale code used by the web experience profile. Example-US.</param>
+        /// <returns>The created web experience profile id.</returns>
+        public string CreateWebExperienceProfile(PaymentConfiguration paymentConfig, BrandingConfiguration brandConfig, string countryIso2Code)
+        {
+            ////Payu does not provide the concept of webprofile
+            ////stored authorization in WebExperienceProfileId
+            return paymentConfig.WebExperienceProfileId;
+        }
+
+        /// <summary>
+        /// Creates a payment transaction and returns the PayUMoney generated payment URL. 
+        /// </summary>
+        /// <param name="returnUrl">The redirect url for PayUMoney callback to web store portal.</param>                
+        /// <param name="order">The order details for which payment needs to be made.</param>        
+        /// <returns>Payment URL from PayUMoney.</returns>
+        public async Task<string> GeneratePaymentUriAsync(string returnUrl, OrderViewModel order)
+        {
+            returnUrl.AssertNotEmpty(nameof(returnUrl));
+            order.AssertNotNull(nameof(order));
+            RemotePost myremotepost = await this.PrepareRemotePost(order, returnUrl);
+            return myremotepost.Post();
+        }
+
+        /// <summary>
+        /// Executes a PayU payment.
+        /// </summary>
+        /// <returns>Capture string id.</returns>
+        public async Task<string> ExecutePaymentAsync()
+        {
+            try
+            {
+                TransactionStatusResponse paymentResponse = await ApiCalls.GetPaymentStatus(this.paymentId);
+                if (paymentResponse != null && paymentResponse.Result.Count > 0 && paymentResponse.Result[0].Status.Equals(Constant.MoneyWithPayU))
+                {
+                    return paymentResponse.Result[0].Amount.ToString();
+                }
+            }
+            catch (PartnerDomainException ex)
```

Comment:
>PartnerDomainException [](start = 19, length = 22)

Dont think the code will throw a PartnerDomainException. Can you verify this case?

Reply:
Since i am not doing anything different for different  kind of exceptions, so catching Exception.

---
In reply to: [183243677](https://github.com/PartnerCenterSamples/Reseller-Web-Application/pull/61#discussion_r183243677) [](ancestors = 183243677)